File: 序号/2610. 转换二维数组.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        # 一维转二维，要求行数尽量少，每行不能重复
        c = Counter(nums)
        ans=[]
        while c:
            keys = list(c)
            ans.append(keys)
            for k in keys:
                if c[k]==1:
                    del c[k]
                else:
                    c[k]-=1
        return ans
                

File: 序号/2610. 转换二维数组.kt
class Solution {
  fun findMatrix(nums: IntArray): List<List<Int>> {
      // 每行数不同
      // 行数尽量少
      // 统计数字出现个数，最多的就是行数
      val d = mutableMapOf<Int,Int>()
      for(x in nums){
          d.merge(x,1,Integer::sum)
      }

      val ans = mutableListOf<List<Int>>()
      while(d.isNotEmpty()){
          ans.add(d.keys.toList())
          for (k in ans.last()){
              d.merge(k,-1){a,b->if(a+b==0)null else a+b}
          }
      }
      return ans
  }
}